### PR TITLE
Fix 769: Rename from <persistent-data-store> to <duckdb-data-store>

### LINKED ISSUE #769: [Lake][OHCLV] Update OHCLV data_factory to use csv_data_writer
### Motivation
OHLCV is still writing/reading from parquet.

We may instead want to use the CSVDataStore + PersistentDataStore to save/read from. But doing this right now is hard, OHLCV code is tightly-bound w/ saving/loading/etc.

### DoD
- [x] OHCLV Data Factory CSV/Saving/Querying code has been abstracted away and is using CSVDataStore + PersistentDataStore
- [x] OHCLV Data Factory and DataStore logic have been separated
- [x] OHCLV Data Factory is querying from PersistentDataStore 
- [x] OHCLV Data Factory tests are up-to-date and every test is passing

------------

### Update According to Trent's comment
- [x] Change the IO methods to a function, in a new module cvsutil.py
- [x] use them into the CSVDataStore class and make it cleaner and thinner

### PR DESCRIPTION
**Pull request by idiom-bytes**
Fixes #769 

Changes proposed in this PR:
- [x] Renamed everything from persistent_data_store -> duckdb_data_store

### CODE REVIEW

File: pdr_backend/lake/etl.py [hunk 1]
```diff
@@ -73,25 +73,25 @@ def _drop_temp_sql_tables(self):
             If exists, drop them and rebuild
         """
         # drop the tables if it exists
-        pds = PersistentDataStore(self.ppss.lake_ss.lake_dir)
+        duckDB = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
```
Comment (calina-c) [inline pdr_backend/lake/etl.py:76]: I don't quite agree to naming this variable duckDB. It's very confusing and error-prone. The library import for DuckDB is duckdb. This variable name is duckDB. To keep the code readable, we need to have a clear distinction on what we are using: the duckdb library or a custom datastore object?

Also, variable names in Python should always be `snake_case`.

I suggest `duckdb_ds` or `duck_ds` or `duck_datastore` for previous instances of `pds`.

File: pdr_backend/lake/test/test_duckdb_data_store.py [hunk 2]
```diff
@@ -150,95 +142,93 @@ def test_fill_table_from_csv(tmpdir):
 
 
 def test_multiton_instances(tmpdir):
-    persistent_data_store_1 = PersistentDataStore(str(tmpdir))
-    persistent_data_store_2 = PersistentDataStore(str(tmpdir))
+    duckdb_1 = DuckDBDataStore(str(tmpdir))
```
Comment (calina-c) [inline pdr_backend/lake/test/test_duckdb_data_store.py:145]: same comment as above, and in other instances

**Review by calina-c — CHANGES_REQUESTED** [review]

File: pdr_backend/lake/test/test_duckdb_data_store.py (hunk 2 above)
Comment (idiom-bytes) [reply pdr_backend/lake/test/test_duckdb_data_store.py:145]: I named things as db in many instances. I'll do that again.

File: pdr_backend/lake/etl.py (hunk 1 above)
Comment (idiom-bytes) [reply pdr_backend/lake/etl.py:76]: just like most other instances, i renamed it simply to "db"

**Pull request merged**